### ISSUE TITLE
ESM module in Zod 2 beta causing runtime error

### ISSUE THREAD

**Issue by lukeramsden** [vriad/zod#215]
With Zod 2.0.0-beta.20:

![Screenshot](https://i.imgur.com/0kYAYgU.png)

With Zod 1.11, no error.

Only seems to happen when using the object type
```ts
// no crash
z.string();
z.date();
// crashes
z.object({});
z.object({
  id: z.string(),
});
```

**Comment by alfonmga** [comment]
It's crashing for me too. Zod works fine on local but when I deploy my Next.js app to production (Vercel) it crashes with the error message: `TypeError: Prototype value can only be an object or null`.

Affected versions: `2.0.0-beta.20` (latest), `2.0.0-beta.19` and `2.0.0-beta.18`.

It works fine on `2.0.0-beta.17`, so looking at the commits pushed after `2.0.0-beta.18` ESM modules changes could be the origin of the issue.

**Comment by o-alexandrov** [comment]
Issue is most probably unrelated to `react native`, but related to ESM introduction only.
Yes, exactly, zod worked fine on `2.0.0-beta.16`, but tried to upgrade to `.20` and saw an issue:
```
Object prototype may only be an Object or null: undefined
```

**Comment by vriad** [comment]
I removed the ESM module in `zod@2.0.0-beta.21`. If you upgrade your builds should work again.

I've also reproduced this issue. Now that I'm aware of this I'll be able to guarantee ESM compatibility before trying to ship an ESM module again. It's not trivial to fix so it'll take some time. There are certain internal circular imports inside Zod's source code that work fine with CommonJS modules but not with ESM.
